src/neml2/base/Factory.cxx

```diff
@@ -26,7 +26,7 @@
 
 namespace neml2
 {
-std::vector<std::string> Factory::pipeline = {"Tensors", "Solvers", "Models", "Drivers"};
+std::vector<std::string> Factory::pipeline = {"Tensors", "Solvers", "Models", "Data", "Drivers"};
```

Comment:
maybe we should move Data in front of Models to reduce the number of stacks.

Reply:
Okay.